core/trino-main/src/main/java/io/trino/operator/TableUpdateOperator.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.Session;
+import io.trino.metadata.Metadata;
+import io.trino.metadata.TableHandle;
+import io.trino.spi.Page;
+import io.trino.spi.PageBuilder;
+import io.trino.spi.block.BlockBuilder;
+import io.trino.spi.predicate.Domain;
+import io.trino.spi.type.Type;
+import io.trino.sql.planner.plan.PlanNodeId;
+
+import java.util.List;
+import java.util.Map;
+import java.util.OptionalLong;
+
+import static com.google.common.base.Preconditions.checkState;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static java.util.Objects.requireNonNull;
+
+public class TableUpdateOperator
+        implements Operator
+{
+    public static final List<Type> TYPES = ImmutableList.of(BIGINT);
+
+    public static class TableUpdateOperatorFactory
+            implements OperatorFactory
+    {
+        private final int operatorId;
+        private final PlanNodeId planNodeId;
+        private final Metadata metadata;
+        private final Session session;
+        private final TableHandle tableHandle;
+        private boolean closed;
+        private final Map<String, Domain> assignments;
+
+        public TableUpdateOperatorFactory(int operatorId, PlanNodeId planNodeId, Metadata metadata, Session session, TableHandle tableHandle, Map<String, Domain> assignments)
+        {
+            this.operatorId = operatorId;
+            this.planNodeId = requireNonNull(planNodeId, "planNodeId is null");
+            this.metadata = requireNonNull(metadata, "metadata is null");
+            this.session = requireNonNull(session, "session is null");
+            this.tableHandle = requireNonNull(tableHandle, "tableHandle is null");
+            this.assignments = requireNonNull(assignments, "assignments is null");
+        }
+
+        @Override
+        public Operator createOperator(DriverContext driverContext)
+        {
+            checkState(!closed, "Factory is already closed");
+            OperatorContext context = driverContext.addOperatorContext(operatorId, planNodeId, TableUpdateOperator.class.getSimpleName());
+            return new TableUpdateOperator(context, metadata, session, tableHandle, assignments);
+        }
+
+        @Override
+        public void noMoreOperators()
+        {
+            closed = true;
```

Comment:
is this threadsafe?

Reply:
To be honest don't know I took this part from TableDeleteOperator,
maybe should I make it volatile?

Reply:
refactored to volatile